### ISSUE TITLE
schema-revision: options/indexingParameters.schema.json

### ISSUE THREAD

**Issue by lzehl** [HumanBrainProject/SANDS#36]
1. I'm not yet happy about decoupling units from the values (holds for all units...) if I remember correctly we can have nested schemas now, so maybe we should always combine those?

**Comment by UlrikeS91** [comment]
Agreed. I would prefer to nest them.

**Comment by lzehl** [comment]
summary of properties:
- units (controlledTerm, count:1) @xgui3783 : keeping this separate from the values only makes sense if we assume the unit is the definition for one step; e.g.:  1 == 1µm)
- axesMin (float, count:1) @xgui3783 : float would make sense to me if we assume here physical units, correct?
- axesMax (float, count:1) @xgui3783 : float would make sense to me if we assume here physical units,  correct?
- steps (integer, count: 1) @xgui3783 : do I assume correctly that this would be pixels?

@UlrikeS91 : too many open issues, let's wait for some clarification.

**Comment by xgui3783** [comment]
@lzehl 

Firstly, maybe not related:

for `indexingParameters.schema.json`, we have:

```json
{
      "@id":{
      "type": "string",
      "description": "Metadata node identifier.",
      "pattern": "sands/options/axes/([A-Za-z0-9_-]+).json"
    }
}
```

but in some others, e.g. `visualizationParameters.schema.json`:

```json
{

    "@id": {
      "type": "string",
      "description": "Metadata node identifier (id).",
      "pattern": "sands/v0.1/core/visualizationParameters/([A-Za-z0-9_-]+).json"
    }
}
```

is it worthwhile standardising the pattern for `@id`, or is this is deliberate choice?

re: issue at hand, this, IMO is very technical, and I question the necessity of it being in SANDS. I vaguely recall why this is necessary, for Lionel spatial search. 

- from a developer perspective, it may not be productive to know how how spatial objects indexed. It makes greater sense to define a constraint (bounding box, mask, etc), and a resolution/threshold (0.5mm). 

- I noticed that `indexingParameters` are used in `axes.schema.json`.indexingParameters . I need to double check with @skoehnen , but I believe there can exist multiple indices on the same space. I also wonder if `axes` is the best place to attach the indexParameter, rather than reference space

**Comment by lzehl** [comment]
@xgui3783 : please ignore the `@id` for now. I'm waiting on some feedback for the pattern from the KG Team and then consistently correct all in all schemas.

As far as I can read from your feedback the technical part needs some more discussion with you, and Stefan and maybe Time?
I would like to get the schemas consistent with what you need from the technical perspective (for the viewer and the spatial search). 

In summary to what we considered so far:
- axes should focus on unitless definitions origin, axes directions
- indexingParameters should focus on unit depending definintions (e.g., resolution which I assumed is defined my min and max plus steps)

I'm happy to do revisions for those schemas. Feel free to get Stefan and Timo on the topic as well.

**Comment by xgui3783** [comment]
@lzehl  I would like to join a meeting with @skoehnen , you and any other stakeholders for sure, before meeting with Timo.

I know you are on holiday, so I think I will have a talk with stefan next week first, and get a technical understanding of the spatial indexing.

PS, aren't you on holiday?

**Comment by lzehl** [comment]
I am on holiday, but pushing SANDS and openMINDS is urgent... We anyway will have a meeting with Timo next week to discuss support for openMINDS and SANDS. It would be great if you me and Stefan could sort out the coordinate, coordinateSpace, axes, indexingParameters, image schemas beforehand.

**Comment by UlrikeS91** [comment]
I updated the schema based on some other things and the wiki in PR #77. Since @lzehl didn't give any instructions on other updates for this schema, I assume that it remained unchanged? 

Please, let me know if something needs to be changed in the PR :)

**Comment by lzehl** [comment]
discussion with @UlrikeS91 @xgui3783 @skoehnen 

This schema is not needed anymore. 
Instead we only capture `voxelSize` and `unit` as optional property in `image.schema.json`.

**Comment by lzehl** [comment]
cf. #34 (this schema will be deleted)
